apollo-router/src/plugins/demand_control/cost_calculator/static_cost.rs

```diff
@@ -321,12 +320,12 @@ impl StaticCostCalculator {
                 list_size_directive.and_then(|dir| dir.size_of(f)),
             ),
             Selection::FragmentSpread(s) => {
-                self.score_fragment_spread(ctx, s, parent_type, list_size_directive)
+                self.score_fragment_spread(ctx, s, list_size_directive)
             }
             Selection::InlineFragment(i) => self.score_inline_fragment(
                 ctx,
                 i,
-                i.type_condition.as_ref().unwrap_or(parent_type),
+                parent_type,
```

Comment:
Moved this into `score_inline_fragment`, I think that makes more sense. It's less brittle if we add more places that call `score_inline_fragment` in the future, as every call site doesn't need to account for the type condition.